src/dns/server.rs

```diff
@@ -1353,27 +1375,50 @@ mod tests {
             }
         }
     }
-    // #[tokio::test]
-    // async fn large_response() {
-    //     // Create and start the proxy with an an empty state. The forwarder is configured to
-    //     // return a large response.
-    //     let addr = new_socket_addr().await;
-    //     let state = new_proxy_state(&[local_workload()], &[], &[]).state;
-    //     let forwarder = Arc::new(FakeForwarder {
-    //         search_domains: vec![],
-    //         ips: HashMap::from([(n("large.com."), new_large_response())]),
-    //     });
-    //     let proxy = DnsProxy::new(addr, NW1, state, forwarder).await.unwrap();
-    //     tokio::spawn(proxy.run());
-    //
-    //     let mut tcp_client = new_tcp_client(addr).await;
-    //
-    //     let resp = send_with_max_size(&mut tcp_client, n("large.com."), RecordType::A, 20).await;
-    //     //resp.answers()[0].
-    //     //let resp = send_request(&mut tcp_client, n("large.com."), RecordType::A).await;
-    //     //assert!(resp.truncated());
-    //     println!("{:?}", resp);
-    // }
+    #[tokio::test]
+    async fn large_response() {
+        initialize_telemetry();
+        // Create and start the proxy with an an empty state. The forwarder is configured to
+        // return a large response.
+        let state = new_proxy_state(&[], &[], &[]);
+        let forwarder = Arc::new(FakeForwarder {
+            search_domains: vec![],
+            ips: HashMap::from([(n("large.com."), new_large_response())]),
+        });
+        let domain = "cluster.local".to_string();
+        let (_signal, drain) = drain::channel();
+        let factory = crate::proxy::DefaultSocketFactory;
+        let server = Server::new(
+            domain,
+            SocketAddr::from(([127, 0, 0, 1], 0)),
+            NW1,
+            state,
+            forwarder,
+            test_metrics(),
+            drain,
+            &factory,
+            true,
+        )
+        .await
+        .unwrap();
+        let tcp_addr = server.tcp_address();
+        let udp_addr = server.udp_address();
+        tokio::spawn(server.run());
+
+        let mut tcp_client = new_tcp_client(tcp_addr).await;
+        let mut udp_client = new_udp_client(udp_addr).await;
+
+        let resp = send_request(&mut tcp_client, n("large.com."), RecordType::A).await;
+        assert!(!resp.truncated(), "TCP should not truncate");
+        assert_eq!(244, resp.answers().len());
+
+        let resp = send_request(&mut udp_client, n("large.com."), RecordType::A).await;
+        //resp.answers()[0].
+        //let resp = send_request(&mut tcp_client, n("large.com."), RecordType::A).await;
+        assert!(resp.truncated());
+        // Not sure we can rely on the truncation being exactly 75, but for now assert
```

Comment:
> Not sure we can rely on the truncation being exactly 75

Why? Genuinely curious.

Also, might be good to add any extra meaningful context in case we have to come back and fix something here with the assert.

Reply:
Wasn't sure if the response size was full deterministic, etc. Seems to be, though